gma/integration_test/src/integration_test.cc

```diff
@@ -902,36 +902,44 @@ TEST_F(FirebaseGmaTest, TestNativeAdLoad) {
   firebase::Future<firebase::gma::AdResult> load_ad_future =
       native_ad->LoadAd(kNativeAdUnit, GetAdRequest());
 
-  WaitForCompletion(load_ad_future, "LoadAd");
-  const firebase::gma::AdResult* result_ptr = load_ad_future.result();
-  ASSERT_NE(result_ptr, nullptr);
-  EXPECT_TRUE(result_ptr->is_successful());
-  EXPECT_FALSE(result_ptr->response_info().adapter_responses().empty());
-  EXPECT_FALSE(
-      result_ptr->response_info().mediation_adapter_class_name().empty());
-  EXPECT_FALSE(result_ptr->response_info().response_id().empty());
-  EXPECT_FALSE(result_ptr->response_info().ToString().empty());
+  // Don't fail loadAd, if NoFill occurred.
+  WaitForCompletionAnyResult(load_ad_future, "LoadAd (ignoring NoFill error)");
+  EXPECT_TRUE(load_ad_future.error() == firebase::gma::kAdErrorCodeNone ||
+              load_ad_future.error() == firebase::gma::kAdErrorCodeNoFill);
+
+  if (load_ad_future.error() == firebase::gma::kAdErrorCodeNone) {
+    const firebase::gma::AdResult* result_ptr = load_ad_future.result();
+    ASSERT_NE(result_ptr, nullptr);
+    EXPECT_TRUE(result_ptr->is_successful());
+    EXPECT_FALSE(result_ptr->response_info().adapter_responses().empty());
+    EXPECT_FALSE(
+        result_ptr->response_info().mediation_adapter_class_name().empty());
+    EXPECT_FALSE(result_ptr->response_info().response_id().empty());
+    EXPECT_FALSE(result_ptr->response_info().ToString().empty());
+
+    // Check image assets.
+    EXPECT_FALSE(native_ad->icon().image_uri().empty());
+    EXPECT_GT(native_ad->icon().scale(), 0);
+    EXPECT_FALSE(native_ad->images().empty());
+
+    // Native ads usually contain only one large image asset.
+    // Check the validity of the first asset from the vector.
+    EXPECT_FALSE(native_ad->images().at(0).image_uri().empty());
+    EXPECT_GT(native_ad->images().at(0).scale(), 0);
+
+    // When the NativeAd is loaded, try loading icon image asset.
+    firebase::Future<firebase::gma::ImageResult> load_image_future =
+        native_ad->icon().LoadImage();
+    WaitForCompletion(load_image_future, "LoadImage");
+    const firebase::gma::ImageResult* img_result_ptr =
+        load_image_future.result();
+    ASSERT_NE(img_result_ptr, nullptr);
+    EXPECT_TRUE(img_result_ptr->is_successful());
+    EXPECT_GT(img_result_ptr->image().size(), 0);
+
+    load_image_future.Release();
+  }
 
```

Comment:
Probably we should LogWarning here about the nofill error, in case someone is running these tests manually.

Reply:
Ack.